pkg/errors/errors.go

```diff
@@ -73,3 +76,35 @@ type ValidationBehavior struct{}
 func (ValidationBehavior) Validation() bool {
 	return true
 }
+
+type SpotPriceError struct {
+	spotPrice        float64
+	currentSpotPrice float64
+	location         string
+	instanceType     string
+}
+
+func NewSpotPriceError(spotPrice, currentSpotPrice float64, location, instanceType string) SpotPriceError {
+	return SpotPriceError{
+		spotPrice:        spotPrice,
+		currentSpotPrice: currentSpotPrice,
+		location:         location,
+		instanceType:     instanceType,
+	}
+}
+
+func (e SpotPriceError) Error() string {
+	return fmt.Sprintf(
+		"spotPrice '%.4f' is too low, current minimum price in region '%s' for instance '%s' is '%.4f'",
```

Comment:
**Question**: wouldn't the error message be better to use location instead of region in case the user specifies a zone?

```suggestion
		"spotPrice '%.4f' is too low, current minimum price for location '%s' and instance '%s' is '%.4f' (zone price for zone locations, maximum of zone prices for regions)",
```

Reply:
Ok, it is clearer this way!